examples/CSVImporter/CSVImporter.go

```diff
@@ -0,0 +1,254 @@
+//     created or updated for this USER_NAME associated with the host_name
+//     provided.
+//     (n.b. - if the USER_NAME already exists on the host_name in question,
+//     specifying these values here will result in the account on host_name
+//     being "taken over" by JumpCloud)
+//
+// 5 - Any invalid combination of values will generate an error and terminate
+//     processing for that entry, but the program will attempt to process the
+//     remainder of the file.
+//
+// 6 - A summary is printed at the conclusion of processing.
+//
+
+
+const (
+    urlBase string = "https://console.jumpcloud.com/api"
+)
+
+
+//
+// Returns the ID for the username specified if it is contained in the
+// list of users provided.  (helper function)
+//
+
+func GetUserIdFromUserName(users []jcapi.JCUser, name string) string {
+    returnVal := ""
+
+    for _, user := range users {
+        if user.UserName == name {
+            returnVal = user.Id
+            break
+        }
+    }
+
+    return returnVal
+}
+
+
+//
+// Main Entry Point...
+//
+
+func main() {
+    // Input parameters
+    var apiKey string
+    var csvFile string
+
+	// Obtain the input parameters
+	flag.StringVar(&csvFile, "csv", "", "-csv=<filename>")
+	flag.StringVar(&apiKey, "key", "", "-key=<API-key-value>")
+	flag.Parse()
+
+    // Attach to JumpCloud
+    jc := jcapi.NewJCAPI(apiKey, urlBase)
+
+	// Setup access the CSV file specified
+	inFile, err := os.Open(csvFile)
+
+    if err != nil {
+        fmt.Println(err)
+        return
+    }
+
+    defer inFile.Close()
+
+    reader := csv.NewReader(inFile)
+    reader.FieldsPerRecord = -1    // indicates records have optional fields
+
+    // Process each user/request record found in the CSV file...
+    recordCount := 0
+
+    for {
+        // Setup work variables
+        var currentUser jcapi.JCUser
+        var currentHost string
+        var currentTag  string
+
+        currentAdmins :=  make(map[string]string)  // "user name", "user id"
+
+        var fieldMap = map[int]*string {
+            0 : &currentUser.FirstName,
+            1 : &currentUser.LastName,
+            2 : &currentUser.UserName,
+            3 : &currentUser.Email,
+            4 : &currentUser.Uid,
+            5 : &currentUser.Gid,
+            // "Sudo" boolean will be handled separately, so no 6
+            7 : &currentUser.Password,
+            8 : &currentHost,
+            9 : &currentTag,
+        }
+
+        // (Re)Fetch all users in JumpCloud (pickup any newly added users)
+        userList, err := jc.GetSystemUsers(false)
+
+        if err != nil {
+            fmt.Printf("Could not read system users, err='%s'\n", err)
+            return
+        }
+
+        // Read next record from CSV file
+    	record, err := reader.Read()
+
+        // Exit loop at the end of file or on error
+    	if err == io.EOF {
+    		fmt.Println("EOF")
+    		break
+    	} else if err != nil {
+    		fmt.Println(err)
```

Comment:
I like to give a sense of what I was doing in these error outputs, as in: Could not read CSV file line #%d, err='%s'